specification/security_model.adoc

```diff
@@ -1,9 +1,438 @@
+    other TVMs or any host software component.
+- Until a TDI is accepted by the TVM:
+  * The TDI is not allowed to DMA into the TVM confidential memory.
+  * Trusted MMIO access to the TDI is blocked by TSM.
+  * Untrusted MMIO access to the TDI may still be allowed by the VMM for
+    ordinary VMs (if and only if the TDI is in the TDISP unlocked state).
+
+By means of the CoVE-IO guest ABI, TVMs are required to explicitly accept
+TDIs into their TCBs. Accepting or rejecting a TDI is a TVM specific decision,
+based on TVM specific set of verification policies and criteria. A TVM accepting
+a TDI does not imply that other TVMs on the same host platform would accept it
+as well.
+
+Multi-function, CoVE-IO-compliant devices may expose multiple TDIs, assigned to
+different TVMs. It is the DSM responsibility to guarantee isolation between all
+assigned TDIs, on a per-TVM basis.
+
+Each TDI must be exclusively assigned to no more than one TVM. However, a single
+TVM can simultaneously be assigned several TDIs, irrespective of whether they
+originate from the same physical device or not.
+
+As a TVM TCB does not include PCIe switches and bridges, a selective PCIe IDE
+stream must be setup to guarantee end-to-end confidentiality and integrity
+protection between a TVM and a TDI. The host VMM is responsible for setting up a
+single selective IDE stream between the host platform and any physical device
+from which one or more TDIs are assigned to TVMs. All TDIs within that physical
+device share the single PCIe IDE stream.
+Since the host VMM is not in the TCB, its role in the stream setup is limited to
+selecting an available stream ID and configuring the device IDE capability. The
+selective IDE stream keys, for both link endpoints, are managed and configured
+by TCB elements (the TSM, DSM and platform RoT).
+
+The main security objective of the CoVE-IO security model is to protect a TVM's
+confidential data integrity and confidentiality while TDIs are assigned to it.
+At the same time, availability of those assigned TDIs is out of this model scope
+as e.g. the host VMM could remove them from the TVM at any time it sees fit.
+Either the DSM or the TDI itself must clear and wipe all TVM confidential data
+in the TDI before the host software stack can fully reclaim an assigned TDI.
+
+== Threat Model
+
+=== Assets
+
+The CoVE-IO security model aims at protecting the following assets:
+
+1. The TVM confidential data, which includes its confidential main memory, code
+   and execution state.
+
+2. An assigned TDI trusted I/O space that is mapped into the TVM address space,
+   also known as a TDI trusted MMIO.
+
+==== Security Objectives
+
+The CoVE-IO security model objectives is to protect the above-described TVM
+assests *confidentiality* and *integrity* from components outside of the TCB.
+
+*Availability* of these assets is out of the CoVE-IO security model scope.
 
 === Adversary Model
```

Comment:
Instead of redefining the adversary model, can't we use a citation to the RISC-V adversary models from the global security model document?

Reply:
We're not really redefining it, but rather contextualizing it for CoVE-IO. Also, we're splitting the software adversary groups between guest and host as they can trigger different attack vectors. And finally, the "startup code adversary" includes both host and device firmware, and implies that device firmware is a platform startup time only adversary. In the CoVE-IO case, device firmware can attack the TVM assets at any point in the platform lifecycle time.
So, to summarize it, I think the RVI security model is accurate but too generic for something like CoVE-IO.